packages/sitecore-jss-nextjs-editing-host/src/editing-middleware.ts

```diff
@@ -0,0 +1,131 @@
+import { parse } from 'url';
+import { Request, Response } from 'express';
+import Server from 'next/dist/next-server/server/next-server';
+import { EditingData } from '@sitecore-jss/sitecore-jss-nextjs';
+import { HtmlProcessor } from './html-processors';
+
+/**
+ * Express middleware for handling requests from the Sitecore Experience Editor.
+ * @constructor
+ * @param {object} nextApp The Next.js app.
+ * @param {string} editRoute The Next.js route to use for rendering.
+ * @param {HtmlProcessor[]} [htmlProcessors] Html processors to run on rendered html.
+ */
+export class EditingMiddleware {
+  constructor(readonly nextApp: Server, readonly editRoute: string, readonly htmlProcessors?: HtmlProcessor[]) {}
+
+  /**
+   * Get Express request handler
+   */
+  public getRequestHandler(): (req: Request, res: Response) => Promise<void> {
```

Comment:
@ambrauer Add JsDoc here, please

Reply:
@sc-illiakovalenko Changed the description a bit, but not sure what else we really need to add.